smartcontracts/contracts/UCamDIDManager.sol

```diff
@@ -1,15 +1,40 @@
 pragma solidity >=0.4.22 <0.6.0;
 
-import './Agentable.sol';
-import './DIDManagerBase.sol';
+import "./Agentable.sol";
+import "./DIDManagerBase.sol";
 
 contract UCamDIDManager is Agentable, DIDManagerBase {
+    constructor(address _dbAddr) public DIDBase(_dbAddr, "did:io:ucam") {}
 
-    constructor(address _dbAddr) DIDBase(_dbAddr, "did:io:ucam") public {}
+    function uint2hex(uint256 i) internal pure returns (bytes memory) {
+        if (i == 0) return bytes("0");
+        uint256 j = i;
+        uint256 length;
+        while (j != 0) {
+            length++;
+            j = j >> 4;
+        }
+        uint256 mask = 15;
+        bytes memory bstr = new bytes(length);
+        uint256 k = length - 1;
+        while (i != 0) {
+            uint256 curr = (i & mask);
+            bstr[k--] = curr > 9
+                ? bytes1(uint8(55 + curr))
+                : bytes1(uint8(48 + curr)); // 55 = 65 - 10
+            i = i >> 4;
+        }
+        return bstr;
+    }
 
     function formDID(bytes20 uid) internal view returns (bytes memory) {
-        // TODO: convert uid to string
-        return abi.encodePacked(db.getPrefix(), uid);
+        bytes memory _uid = "";
+
+        for (uint8 i = 0; i < uid.length; i++) {
+            _uid = abi.encodePacked(_uid, uint2hex(uint8(uid[i])));
+        }
```

Comment:
may not be efficient, try this one https://ethereum.stackexchange.com/questions/8346/convert-address-to-string